File: src/main/java/META-INF/resources/dataforms/field/common/AudioField.js
/**
 * @fileOverview {@link AudioField}クラスを記述したファイルです。
 */

'use strict';

import { StreamingField } from './StreamingField.js';


/**
 * @class AudioField
 * 音声ファイルアップロードフィールドクラス。
 * @extends StreamingField
 */
export class AudioField extends StreamingField {
	/**
	 * HTMLエレメントとの対応付けを行います。
	 * <pre>
	 * 削除チェックボックス、ダウンロードリンクなどの設定を行います。
	 * </pre>
	 */
	attach() {
		super.attach();
	}
}

File: src/main/java/jp/dataforms/fw/field/common/AudioField.java
package jp.dataforms.fw.field.common;

import jp.dataforms.fw.dao.file.AudioData;
import jp.dataforms.fw.dao.file.FileObject;

/**
 * 音声フィールドクラス。
 *
 */
public class AudioField extends StreamingField<AudioData> {

	/**
	 * コンストラクタ。
	 *
	 */
	public AudioField() {
		super(null);
	}

	/**
	 * コンストラクタ。
	 *
	 * @param id フィールドID。
	 */
	public AudioField(final String id) {
		super(id);
	}

	@Override
	protected void onBind() {
		super.onBind();
	}

	@Override
	public void init() throws Exception {
		super.init();
		this.setAdditionalHtml(this.getPage().getPageFramePath() + "/AudioField.html");
	}

	@Override
	protected FileObject newFileObject() {
		return new AudioData();
	}



}
